dotcom-rendering/src/model/insert-newsletter-signup.ts

```diff
@@ -0,0 +1,167 @@
+interface PlaceInArticle {
+	index: number;
+	distance: number;
+	afterText: boolean;
+	afterSupporting: boolean;
+}
+
+const isCloseAfterSupportingElement = (
+	index: number,
+	elements: CAPIElement[],
+): boolean => {
+	const elementIsSupporting = (element?: CAPIElement): boolean =>
+		!!(element && 'role' in element && element.role === 'supporting');
```

Comment:
Unfortunately, that would give us a typeError here - `element`(if defined) is a `CAPIElement` and not every type in the Union has the "role" property. 
It works on line 22 with "_type" because every `CAPIElement` has a "_type".